python/lsst/daf/persistence/policy.py

```diff
@@ -0,0 +1,392 @@
+        return self
+
+    def __initFromYamlFile(self, path):
+        """Opens a file at a given path and attempts to load it in from yaml.
+
+        :param path:
+        :return:
+        """
+        f = file(path, 'r')
+        self.__initFromYaml(f)
+
+    def __initFromYaml(self, stream):
+        """Loads a YAML policy from any readable stream that contains one.
+
+        :param stream:
+        :return:
+        """
+        # will raise yaml.YAMLError if there is an error loading the file.
+        self.data = yaml.load(stream)
+        return self
+
+    def __getitem__ (self, name):
+        data = self.data
+        for key in name.split('.'):
+            if key in data:
+                data = data[key]
+            else:
+                return None
+        if isinstance(data, collections.Mapping):
+            data = Policy(data)
+        return data
+
+    def __setitem__(self, name, value):
+        if isinstance(value, collections.Mapping):
+            keys = name.split('.')
+            d = {}
+            cur = d
+            for key in keys[0:-1]:
+                cur[key] = {}
+                cur = cur[key]
+            cur[keys[-1]] = value
+            self.update(d)
+        data = self.data
+        keys = name.split('.')
+        for key in keys[0:-1]:
+            data = data.setdefault(key, {})
+        data[keys[-1]] = value
+
+    def __contains__(self, key):
+        d = self.data
+        keys = key.split('.')
+        for k in keys[0:-1]:
+            if k in d:
+                d = d[k]
+            else:
+                return False
+        return keys[-1] in d
+
+    def update(self, other):
+        """Like dict.update, but will add or modify keys in nested dicts, instead of overwriting the nested
+        dict entirely.
+
+        For example, for the given code:
+        foo = {'a': {'b': 1}}
+        foo.update({'a': {'c': 2}})
+
+        If foo is a dict, then after the update foo == {'a': {'c': 2}}
+        But if foo is a Policy, then after the update foo == {'a': {'b': 1, 'c': 2}}
+        """
+        def doUpdate(d, u):
+            for k, v in u.iteritems():
+                if isinstance(d, collections.Mapping):
+                    if isinstance(v, collections.Mapping):
+                        r = doUpdate(d.get(k, {}), v)
+                        d[k] = r
+                    else:
+                        d[k] = u[k]
+                else:
+                    d = {k: u[k]}
+            return d
+        doUpdate(self.data, other)
+
+    def merge(self, other):
+        """Like Policy.update, but will add keys & vales from other that DO NOT EXIST in self. Keys and
+        values that already exist in self will NOT be overwritten.
+
+        :param other:
+        :return:
+        """
+        other.update(self)
+        self.data = other.data
+
+    def getSubPolicyVal(self, subPolicyName, subPolicyKey):
+        """Look for a subpolicy and if found, return the value for a key.
```

Comment:
Where is this used?  I don't recall needing something like this, and it seems a bit dangerous as you don't know (and can't tell afterwards) where in the hierarchy it will be found.